plugins/modules/idrac_system_erase.py

```diff
@@ -0,0 +1,569 @@
+
+#
+# Dell OpenManage Ansible Modules
+# Version 9.7.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: idrac_system_erase
+short_description: Erase system and storage components of the server
+version_added: "9.7.0"
+description:
+  - This module allows erasing System components such as iDRAC, BIOS, DIAG,
+    etc., and Storage components such as PERC NV cache, Non-volatile memory,
+    Cryptographic erase PD, etc. of the server.
+extends_documentation_fragment:
+  - dellemc.openmanage.idrac_x_auth_options
+options:
+  component:
+    description:
+      - System and storage components to erase.
+      - Below are the supported components.
+        AllApps
+        BIOS
+        CryptographicErasePD
+        DIAG
+        DPU
+        DrvPack
+        IDRAC
+        LCData
+        NonVolatileMemory
+        OverwritePD
+        PERCNVCache
+        ReinstallFW
+        vFlash
+    type: list
+    elements: str
+    required: true
+  power_on:
+    description:
+      - This allows to power on the server after erase operation. This is applicable only when I(job_wait) is C(true).
+      - C(true) will power on the server.
+      - C(false) will not power on the server.
+    type: bool
+    default: false
+  job_wait:
+    description:
+      - Whether to wait till completion of the job. This is applicable when I(power_on) is C(true).
+      - C(true) will wait for job completion.
+      - C(false) will not wait for job completion.
+    type: bool
+    default: true
+  job_wait_timeout:
+    description:
+      - The maximum wait time of I(job_wait) in seconds. The job is tracked only for this duration.
+      - This option is applicable when I(job_wait) is C(true).
+    type: int
+    default: 1200
+  resource_id:
+    description:
+      - Manager ID of the iDRAC.
+    type: str
+requirements:
+  - "python >= 3.9.6"
+author:
+  - "Rajshekar P(@rajshekarp87)"
+attributes:
+    check_mode:
+        description: Runs task to validate without performing action on the target machine.
+        support: full
+    diff_mode:
+        description: Runs the task to report the changes made or to be made.
+        support: none
+notes:
+    - Run this module from a system that has direct access to Dell iDRAC.
+    - This module supports only iDRAC9 and above.
+    - This module supports IPv4 and IPv6 addresses.
+"""
+
+EXAMPLES = r"""
+---
+- name: Erase single component and power on the server
+  dellemc.openmanage.idrac_system_erase:
+    idrac_ip: 198.162.0.1
+    idrac_user: username
+    idrac_password: passw0rd
+    ca_path: "/path/to/ca_cert.pem"
+    component: ["BIOS"]
+    power_on: true
+
+- name: Erase multiple components and don't power on the server after erase operation
```

Comment:
- name: Delete multiple servers or storage components and do not power on the server after the delete operation is completed.